xarray/test/test_backends.py

```diff
@@ -492,6 +492,21 @@ def create_tmp_file(suffix='.nc', allow_cleanup_failure=False):
             if not allow_cleanup_failure:
                 raise
 
+@contextlib.contextmanager
+def create_tmp_files(nfiles, suffix='.nc', allow_cleanup_failure=False):
```

Comment:
Thanks @shoyer, this makes the code cleaner.  However, it may be operating slower because there could be more contexts in play.  It does remove redundant code, which is always a plus.